src/Http/Client/Cookie/RequestCookies.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.5.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Client\Cookie;
+
+use InvalidArgumentException;
+use Iterator;
+use Psr\Http\Message\ServerRequestInterface;
+
+class RequestCookies extends CookieCollection implements Iterator
```

Comment:
Would extending IteratorAggregate or ArrayIterator make sense here? Seems like we might be able to save some code by using those.